src/core/annotation.js

```diff
@@ -2552,6 +2552,36 @@ class InkAnnotation extends MarkupAnnotation {
         });
       }
     }
+
+    if (!this.appearance) {
+      // The default stroke color is black.
+      const strokeColor = this.color
+        ? Array.from(this.color).map(c => c / 255)
+        : [0, 0, 0];
+
+      const borderWidth = this.borderStyle.width || 1;
+
+      this._setDefaultAppearance({
+        xref: parameters.xref,
+        extra: `${borderWidth} w`,
```

Comment:
After reading the specs for Ink annotation, I think you forgot to handle dash patterns.

Reply:
In this patch I simply implemented the bare minimum required to fix the referenced issue, and I think that we'd need *real world* test-cases to be able to implement dash patterns correctly here (and that that could perhaps be done in a follow-up instead).

Reply:
Looking at the PDF document in the issue, it contains a `[3]` dashArray (i.e. the default value) which according to https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/PDF32000_2008.pdf#G7.3881971 thus would suggest that there should be gaps in the rendered InkAnnotation.

*However*, if you actually look at the rendering and/or Appearance stream that Adobe Reader generates for that PDF document (or even the actual rendering in PDFium), there's *no* dashes present and only solid lines being rendered.
Furthermore, I've tried modifying the reduced test-case included in this patch to specify a non-standard dashArray, and both Adobe Reader and PDFium seem to outright ignore those values and only render the InkAnntations with solid lines.

All-in-all, ignoring the dashArray seems OK in practice here as far as I can tell.

Reply:
@Snuffleupagus if you want I can try to find usage of dash patterns in my corpus of PDFs, I "just" need a way to detect them (like https://github.com/marco-c/pdf-finder/blob/f045328bbda0bf3a371e992e5c843ba23c02d502/run.py#L20-L37).